### DISCUSSION TITLE
Keeping up with Gradle modularisation graph stats - Android

### DISCUSSION THREAD

**Discussion by ritesh-singh** [MobileNativeFoundation/discussions#193] [General]
Hi All,

I wanted to start a conversation regarding Gradle module graphs. Keeping track of the module graph is an important aspect of maintaining a modular codebase, and using a Gradle plugin like [modules-graph-assert](https://github.com/jraska/modules-graph-assert) to generate a `.dot` file that can be analyzed by tools like [Graphviz](https://graphviz.org/) or [Gephi](https://gephi.org/) is a good way to visualize the module graph and identify potential issues.

Analyzing the degree, in-degree, modularity and other statistics of the graph can help us identify modules that have too many dependencies, or modules that are highly interconnected with other modules. This can be a sign that these modules are becoming too complex and may need to be refactored.

These tools can help us visualize the module graph and can also help us identify clusters of modules that are highly connected, which can be an indicator of strong cohesion and well-designed modules.

Making use of  `graph-assert` gradle plugin and gephi open source api's  - for generating graph stats like degree, in-degree etc.. and determine which node/module needs to be looked at?

What are your thoughts? Any one tried or have similar set-up in place in their org?

**Comment by waliahimanshu** [comment]
I have not come across another setup, but we are also in process of refactoring highly coupled modules.
And then enforcing the `allowed module` regex using the `graph-assert` is the way to make sure the graph height remains in check, in addition to blocks direct referencing between feature modules etc.
